### PR TITLE
pandas must be an optional requirement 

### PR DESCRIPTION
**Pull request by dmonroy**
pandas must be an optional requirement as no all applications that uses scrunch needs pandas.

Also pandas requires numpy and having both wastes unnecessary resources when building an environment from scratch.

As soon as this is merged I'll add the corresponding instructions to the wiki for the people needing pandas.

### CODE REVIEW

File: scrunch/datasets.py [hunk 1]
```diff
@@ -2013,6 +2019,8 @@ def forks_dataframe(self):
             A DataFrame representation of all attributes from all forks
             on the given dataset.
         """
+        if pd is None:
+            raise ModuleNotFoundError("No module named 'pandas'")
```
Comment (jjdelc) [inline scrunch/datasets.py]: This message should read more like "Pandas is not installed, please install it in your environment to use this function" or something along those lines.

File: setup.py [hunk 2]
```diff
@@ -60,7 +59,7 @@
 
             # local
         ],
-        'pandas': ['pandas'],
+        'pandas': ['pandas<0.20'],
```
Comment (jjdelc) [inline setup.py]: Why the lock on this version?

File: scrunch/datasets.py (hunk 1 above)
Comment (dmonroy) [reply scrunch/datasets.py]: I was thinking on replicate the error that python raises when you import something that is not installed, but your message makes more sense

File: setup.py (hunk 2 above)
Comment (dmonroy) [reply setup.py]: when we remove that version constraint tests fails with: ```E           AttributeError: 'DataFrame' object has no attribute 'sort'```

Comment (dmonroy) [reply setup.py]: and fixed it with `_forks.sort_values(by=['creation_time'], inplace=True)`

**Comment by jjdelc** [comment]
Can you add a test verifying this behavior?

**Comment by dmonroy** [comment]
@jjdelc what do you think now about this PR? 

pandas (and numpy) adds +200MB to our builds and lot of processing time to the pipeline

**Pull request merged**